lightning/src/ln/channel.rs

```diff
@@ -5455,13 +5458,12 @@ impl<'a, Signer: Sign, K: Deref> ReadableArgs<(&'a K, u32)> for Channel<Signer>
 			config.as_mut().unwrap().forwarding_fee_proportional_millionths = Readable::read(reader)?;
 			config.as_mut().unwrap().cltv_expiry_delta = Readable::read(reader)?;
 			config.as_mut().unwrap().announced_channel = Readable::read(reader)?;
-			config.as_mut().unwrap().commit_upfront_shutdown_pubkey = Readable::read(reader)?;
+			// config.as_mut().unwrap().commit_upfront_shutdown_pubkey = Readable::read(reader)?;
 		} else {
 			// Read the 8 bytes of backwards-compatibility ChannelConfig data.
 			let mut _val: u64 = Readable::read(reader)?;
 		}
-
-		let channel_id = Readable::read(reader)?;
+		let handshake_config = Some(ChannelHandshakeConfig::default());		let channel_id = Readable::read(reader)?;
```

Comment:
Note here that you're creating a fresh `handshake_config` with default values, never reading any data to update it, then later setting `commit_upfront_shutdown_pubkey` with the value from the fresh object.